### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by musangamfure**
**Hello,** 

In this PR, I have worked on reducer and action features for the Bookstore project.

**Here are what I have implemented:**

- [ ] Add Redux Toolkit `(npm install react-redux @reduxjs/toolkit).`
- [ ] Create a directory that will contain all your Redux logic `(/src/redux)`
- [ ] Configure a Redux store `(/src/redux/store.js)`
- [ ] Define a slice of state for books that:
      1. Includes an array of books (initial state: empty array)
      2.  Includes a reducer that adds a book
      3. Includes a reducer that removes a book
- [ ] Define a slice of state for categories that:
      1. Includes an array of categories (initial state: empty array)
      2. Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)
- [ ] Structure my application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure

Kindly Check the code and let me know what you think.

Happy coding🎉🎉

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: 'Under construction',
+  reducers: {
+    // No action required as the initial state is already set
+  },
```
Comment (codepantha) [inline src/redux/categories/categoriesSlice.js]: - [ ] According to project requirements, please add a reducer that checks the status and always returns 'Under construction'
**HINT:**👇
```js
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```

- [ ] I can see that your initialState is 'Under construction'. Please ensure that you create an initialState with a state value `categories` that contains an empty array as given in the project requirements 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @musangamfure 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Add reducers and actions 📌 
✔️ Good commit history

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (rloterh) [reply src/redux/categories/categoriesSlice.js]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Emmanuel,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
- Good job implementing recommended changes!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (musangamfure) [reply src/redux/categories/categoriesSlice.js]: Hello @rloterh 

Thank you so much for your nice comments and approving my project.

**Pull request merged**